### PR TITLE
Feat/prompter midi pedal

### PR DESCRIPTION
**Pull request by jesperstarkar**
* **What kind of change does this PR introduce?** 
Feature. Adds WebMidi support for controlling the prompter. Intended to be used with any midi foot pedals. Also, adds various improvements and restructuring of the prompter control-code.

* **What is the current behavior?**
This PR adds a new feature.

* **What is the new behavior (if this is a feature change)?**
- Plug n play support for WebMidi interfaces.
- Listens to CC messages on channel 8, the default channel for expression pedals (recommended device).
- Exposes parameters to the URL to calibrating how the input range maps to the control signal for the prompter.
- Exposes a debug mode to help understanding the midi signal and the control signal, so you can read, adjust and re-iterate quickly in a studio, collaborating with a presenter testing and validating the pedal control.

* **Other information**:
- The midi pedal controller is now the template for how other devices should be written.

**Status**

- [ ] Code documentation for the relevant parts in the code have been added/updated by the PR author
- [x] The functionality has been tested by the PR author
- [ ] The functionality has been tested by NRK

### CODE REVIEW

File: meteor/client/ui/Prompter/controller/midi-pedal-device.ts [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+		}
+		if (this.rangeFwdMax <= this.rangeNeutralMax) {
+			console.error('rangeFwdMax must be larger to rangeNeutralMax. Pedal control will not initialize')
+			return
+		}
+
+		webmidi.enable(this.setupMidiListeners.bind(this))
+	}
+
+	public destroy() {
```
Comment (jstarpl) [inline meteor/client/ui/Prompter/controller/midi-pedal-device.ts:48]: Shouldn't this run `webmidi.disable()` to clean up and free up any MIDI resources we have connected to?

**Review by jstarpl — CHANGES_REQUESTED** [review]
Just a single comment regarding the `destroy()` method.

File: meteor/client/ui/Prompter/controller/midi-pedal-device.ts (hunk 1 above)
Comment (jesperstarkar) [reply meteor/client/ui/Prompter/controller/midi-pedal-device.ts:48]: Added

**Comment by jesperstarkar** [comment]
Cleaned up.

I _think_ the loops around the RAF are safe, but I'm not sure.
I see memory grow as I run the pedal, but I'm not sure if that is just normal behaviour or a leakage.

**Comment by jesperstarkar** [comment]
@jstarpl care to add the interpolation?

**Review by jstarpl — APPROVED** [review]
Looks good to me!

**Comment by jesperstarkar** [comment]
Functionality is in place, code looks good.

Documentation is not done.

**Comment by jesperstarkar** [comment]
@jstarpl can the debug box obey the margin that's being set?

**Comment by jesperstarkar** [comment]
Description added

**Comment by jstarpl** [comment]
> 
> 
> @jstarpl can the debug box obey the margin that's being set?

Done.

**Comment by sparkpunkd** [comment]
Took a lot just post merge. Looks good. 
Confirmed new dependencies are OK.

**Pull request merged**